### DISCUSSION TITLE
Defining an override in a  C# project file is also adding it to other C# project files that I don't want them added to.

### DISCUSSION THREAD

**Discussion by Alriightyman** [premake/premake-core#2213] [Q&A] [answered]
I have 2 C# projects - one that needs an extra bit in a property group and the other does not.  However, even though the override is only defined in one of the project files, it always applies to the other project too.  How can I tell premake to only use it in the project file its defined in?

for reference:
`	premake.override(premake.vstudio.dotnetbase, "projectProperties", function(base, prj)
		_p(1,'<PropertyGroup>')
		premake.w('Extra added property stuff here')
		_p(1,'</PropertyGroup>')
		base(prj)
	end)`

**Comment by samsinsane** [comment, marked as answer]
You can either register an API and use that, which would give you the most flexibility to enable this override for any project you want. Or, you can just wrap the extra output in an `if prj.name == "MyProj" then ... end`. For examples of how APIs are registered, see [this file](https://github.com/premake/premake-core/blob/master/src/_premake_init.lua) which contains most of the APIs in Premake, then you would use `myapi "Value"` in the project configuration, and then the conditional in your override would be `if prj.myapi == "Value" then`.

**Reply by Alriightyman** [reply]
Thanks!  Registering an API worked perfectly.
